File: IO/FileHelper.vb
Imports System.IO

Namespace kCura.WinEDDS.IO
	Public Class FileHelper
		Implements IFileHelper

		Public Function Create(filePath As String) As FileStream Implements IFileHelper.Create
			return File.Create(filePath)
		End Function

		Public Sub Delete(filePath As String) Implements IFileHelper.Delete
			File.Delete(filePath)
		End Sub

		Public Sub Copy(sourceFilePath As String, destinationFilePath As String) Implements IFileHelper.Copy
			Copy(sourceFilePath, destinationFilePath, false)
		End Sub

		Public Sub Copy(sourceFilePath As String, destinationFilePath As String, overwrite As Boolean) Implements IFileHelper.Copy
			File.Copy(sourceFilePath, destinationFilePath, overwrite)
		End Sub

		Public Function Exists(filePath As String) As Boolean Implements IFileHelper.Exists
			return File.Exists(filePath)
		End Function

		Public Function Create(filePath As String, append As Boolean) As FileStream Implements IFileHelper.Create
			Dim mode As FileMode = If(append, FileMode.Append, FileMode.Create)
			return new FileStream(filePath, mode, FileAccess.ReadWrite, FileShare.None)
		End Function
	End Class
End NameSpace

File: IO/FileHelper.cs
using System.IO;

namespace kCura.WinEDDS.Core.IO
{
	public class FileHelper : IFileHelper
	{
		public FileStream Create(string filePath)
		{
			return File.Create(filePath);
		}

		public void Delete(string filePath)
		{
			File.Delete(filePath);
		}

		public void Copy(string sourceFilePath, string destinationFilePath)
		{
			Copy(sourceFilePath, destinationFilePath, false);
		}

		public void Copy(string sourceFilePath, string destinationFilePath, bool overwrite)
		{
			File.Copy(sourceFilePath, destinationFilePath, overwrite);
		}

		public bool Exists(string filePath)
		{
			return File.Exists(filePath);
		}
	}
}